### PR TITLE
Cews twitter

### PR DESCRIPTION
**Pull request by cews7**
@scmountain please read through each file change to get a better sense of how this works. 

In a nutshell... 

- I'm using the twitter gem to grab the user's data once he is authenticated through twitter-omniauth. 
- The service file is the only thing that may look a bit weird compared to what you are used to seeing in the constructor method. 

- For the time being to demonstrate it works, I threw the sign in, sign out functionality into the application layout file. 

DM me with questions as needed.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
   root "home#show"
 
+  get '/auth/twitter', as: :login
```
Comment (CjMoore) [inline config/routes.rb:6]: Just so I understand. When the user is directed to this route, the oauth process starts automatically, and the callback is directed to the sessions controller where the user and session is created? @cews7

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class SessionsController < ApplicationController
+  def create
+    auth = request.env["omniauth.auth"]
```
Comment (CjMoore) [inline app/controllers/sessions_controller.rb:3]: this auth variable is the data thats returned from the oauth callback?

File: config/routes.rb (hunk 1 above)
Comment (cews7) [reply config/routes.rb:6]: yup

File: app/controllers/sessions_controller.rb (hunk 2 above)
Comment (cews7) [reply app/controllers/sessions_controller.rb:3]: yup

**Review by CjMoore — APPROVED** [review]
Looks good. I can swap out the placeholders after all this stuff is merged.

File: app/models/services/twitter_service.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'twitter'
+
+class TwitterService
+  attr_reader :client, :nickname
+  def initialize(user)
+    @nickname    = user.nickname
+    @avatar      = user.image
```
Comment (scmountain) [inline app/models/services/twitter_service.rb:7]: yay for not having to use AWS!

**Review by scmountain — APPROVED** [review]

**Pull request merged**